### DISCUSSION TITLE
No Pin PULL_DOWN?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#16458] [ESP8266]
No Pin PULL_DOWN?  Using MicroPython v1.24.0 on 2024-10-25, Platform 'ESP module with ESP8266'

```
>>> from machine import Pin
>>> dir(Pin)
['__class__', '__name__', 'value', 'IN', 'IRQ_FALLING', 'IRQ_RISING', 'OPEN_DRAIN', 'OUT', 'PULL_UP', '__bases__', '__dict__', 'init', 'irq', 'off', 'on']
```

**Comment by projectgus** [comment]
The ESP8266 hardware only supports pull-ups not pull-downs.

**Comment by Jibun-no-Kage** [comment]
Between the 'A.I.' example which was wrong... if not humorous... the documentation does not make this clear at all.  It does qualify well the ESP32 limitations most of the time, but not for ESP8266 on this point.

**Reply by projectgus** [reply]
That's fair about the docs. There are unfortunately a lot of these minor differences between ports that aren't clearly mentioned. I'll make a note to update the docs for the pins.
